strsignal.c is only compiled for the tests, but fails with:

strsignal.c:66:31: error: invalid application of 'sizeof' to incomplete type 'const char * const[]'
  if (s > 0 && s < (int)(sizeof(sys_siglist) / sizeof(sys_siglist[0])))
diff -u -r ../dpkg-1.17.11/lib/compat/strsignal.c ./lib/compat/strsignal.c
--- ../dpkg-1.17.11/lib/compat/strsignal.c	2014-08-09 22:03:58.000000000 +0200
+++ ./lib/compat/strsignal.c	2014-08-13 20:57:32.998986392 +0200
@@ -63,9 +63,10 @@
 {
 	static char buf[100];
 
+#ifndef __ANDROID__
 	if (s > 0 && s < (int)(sizeof(sys_siglist) / sizeof(sys_siglist[0])))
 		return sys_siglist[s];
-
+#endif
 	sprintf(buf, _("Unknown signal %d"), s);
 
 	return buf;
